Client/Frontend/Browser/TopTabsViewController.swift

```diff
@@ -71,6 +71,9 @@ class TopTabsViewController: UIViewController {
         return delegate
     }()
     
+    private weak var lastNormalTab: Tab?
+    private weak var lastPrivateTab: Tab?
```

Comment:
Now that we are caching the last selected tabs for normal/private mode in the VC, we'll want to make sure to respond and act on all possible tab manipulation events. For example, when selecting 'Close all tabs' I don't thin the `didSelectedTabChange` delegate gets fired but the `didRemoveTab` one does so we'll want to make sure to implement that as well.